### ISSUE TITLE
Wrong order of padding values in crop function

### ISSUE THREAD

**Issue by Fleyderer** [pytorch/vision#7997]
### 🐛 Describe the bug

I've found bug in `torchvision -> transforms -> _functional_tensor.py -> def crop(...)`
Padding values has `L, T, R, B` order, but function pad uses `L, R, T, B` order.
So when I have to pad right part of image, function `resized_crop` adds padding on top of it, because it uses function `crop`

Source code:
```
def crop(img: Tensor, top: int, left: int, height: int, width: int) -> Tensor:
    _assert_image_tensor(img)

    _, h, w = get_dimensions(img)
    right = left + width
    bottom = top + height

    if left < 0 or top < 0 or right > w or bottom > h:
        padding_ltrb = [ 
            max(-left + min(0, right), 0),
            max(-top + min(0, bottom), 0),
            max(right - max(w, left), 0),
            max(bottom - max(h, top), 0),
        ] # <-- here
        return pad(img[..., max(top, 0) : bottom, max(left, 0) : right], padding_ltrb, fill=0)
    return img[..., top:bottom, left:right]
```

I have to mention that this bug is in `center_crop` function too (Just search for "padding_ltrb" to find all wrong ordered lists of padding.

### Versions

torchvision version: 0.15.2+cu118

**Comment by NicolasHug** [comment]
Thanks for the report @Fleyderer . I agree that seeing `padding_ltrb` here is a bit surprising. That being said, results look OK:

```py
import torch
from torchvision.transforms import functional as F

img = torch.rand(3, 10, 10)
out = F.crop(img, 5, 5, 10, 15)
out.shape
# %%
out[0]
```

```
tensor([[0.7061, 0.1831, 0.4297, 0.9661, 0.4423, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.8602, 0.8052, 0.6932, 0.4612, 0.7612, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.8547, 0.5132, 0.1214, 0.6592, 0.6851, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.4385, 0.4716, 0.0876, 0.4421, 0.9476, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.6609, 0.0488, 0.2011, 0.3065, 0.4809, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
        [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000]])
```

I can't reproduce the "padding on top" you mentioned. Would you have a reproducing example?

Note that the image that gets pad is `img[..., max(top, 0) : bottom, max(left, 0) : right]`, not `img`, which should prevent any padding at the top of the image

**Comment by Fleyderer** [comment]
Hello @NicolasHug, thanks for reply! I've found problem: in my project I've changed some source code to make my module traceable (`torch.jit.trace(...)`) and removed `_parse_pad_padding` call and some others to make things work.

Source code of `_parse_pad_padding` :
```
def _parse_pad_padding(padding: Union[int, List[int]]) -> List[int]:
    if isinstance(padding, int):
        if torch.jit.is_scripting():
            # This maybe unreachable
            raise ValueError("padding can't be an int while torchscripting, set it as a list [value, ]")
        pad_left = pad_right = pad_top = pad_bottom = padding
    elif len(padding) == 1:
        pad_left = pad_right = pad_top = pad_bottom = padding[0]
    elif len(padding) == 2:
        pad_left = pad_right = padding[0]
        pad_top = pad_bottom = padding[1]
    else:
        pad_left = padding[0]
        pad_top = padding[1]
        pad_right = padding[2]
        pad_bottom = padding[3]

    return [pad_left, pad_right, pad_top, pad_bottom]
```

Threre is reordering logic in this function. 
BTW this logic looks ugly: firstly we make LTRB padding and then always (!) swap it to LRTB.

Maybe this code is need to be changed...

**Comment by pmeier** [comment]
There is an explanation of this behavior in the transforms v2 API:

https://github.com/pytorch/vision/blob/1a9ff0d728787df66a372283328158879805dcad/torchvision/transforms/v2/functional/_geometry.py#L1156-L1158

So unfortunately, we are stuck with both variants :confounded: This also creates "issues" internally:

https://github.com/pytorch/vision/blob/1a9ff0d728787df66a372283328158879805dcad/torchvision/transforms/v2/_geometry.py#L886-L887

I remember this costing me quite some time to figure out.
